File: 242-ValidAnagram/242-ValidAnagram.cs
// Last updated: 12/11/2025, 8:02:06 PM
public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary<char, int> hashMap = new();

        if(s.Length != t.Length)
        {
            return false;
        }

        for(int i=0; i<s.Length; i++)
        {
            hashMap.TryAdd(s[i], 0);
            hashMap[s[i]] += 1;
        }

        for(int i=0; i<t.Length; i++)
        {
            if(!hashMap.ContainsKey(t[i]))
            {
                return false;
            }
            hashMap[t[i]] -= 1;
            if(hashMap[t[i]] == 0)
            {
                hashMap.Remove(t[i]);
            }
        }
        return true;
    }
}

File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 12/11/2025, 8:02:03 PM
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        anagramDict = {}

        if len(s) != len(t):
            return False
        for char in s:
            if char in anagramDict:
                anagramDict[char] += 1
            else:
                anagramDict[char] = 1
        
        for char in t:
            if anagramDict.get(char) == None:
                return False
            else:
                anagramDict[char] -= 1
                if anagramDict[char] == 0:
                    del anagramDict[char]
        return True if len(anagramDict) == 0 else False
